targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/us_ticker.c

```diff
@@ -0,0 +1,346 @@
+         * may be set while TMR1.CURCNT is being read.  We don't know if the pend bit matches the TMR1 state.
+         * To prevent this, the pending bit is read twice, and we see if it matches; if it doesn't, loop around again.
+         *
+         * Note the TMR1 interrupt is enabled on each iteration of the loop to flush out any pending TMR1 interrupt,
+         * thereby clearing any TMR1 pend's.  This have no effect if this routine is called with interrupts globally disabled.
+         */
+
+        NVIC_DisableIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);		// Prevent Upper_count increment
+        tmrpend0 = NVIC_GetPendingIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);
+        // Check if there is a pending interrupt for timer 1
+
+        __DMB();                                                    // memory barrier: read GP0 before GP1
+
+        tmrcnt0 = adi_tmr_registers[ADI_TMR_DEVICE_GP0]->CURCNT;    // to minimize skew, read both timers manually
+
+        __DMB();                                                    // memory barrier: read GP0 before GP1
+
+        tmrcnt1 = adi_tmr_registers[ADI_TMR_DEVICE_GP1]->CURCNT;    // read both timers manually
+
+        totaltmr0 = tmrcnt0;        								// expand to u32 bits
+        totaltmr1 = tmrcnt1;        								// expand to u32 bits
+
+        tmrcnt0 &= 0xff00u;
+        tmrcnt1 <<= 8;
+
+        __DMB();
+
+        uc1 = *ucptr;												// Read Upper_count
+
+        tmrpend1 = NVIC_GetPendingIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);
+        // Check for a pending interrupt again.  Only leave loop if they match
+
+        NVIC_EnableIRQ(adi_tmr_interrupt[ADI_TMR_DEVICE_GP1]);		// enable interrupt on every loop to allow TMR1 interrupt to run
+    } while ((tmrcnt0 != tmrcnt1) || (tmrpend0 != tmrpend1));
+
+    totaltmr1 <<= 8;                 // Timer1 runs 256x slower
+    totaltmr1 += totaltmr0 & 0xffu;  // Use last 8 bits of Timer0 as it runs faster
+    // totaltmr1 now contain 24 bits of significance
+
+    if (tmrpend0) {					 // If an interrupt is pending, then increment local copy of upper count
+        uc1++;
+    }
+
+    uint64_t Uc = totaltmr1;         // expand out to 64 bits unsigned
+    Uc += ((uint64_t) uc1) << 24;    // Add on the upper count to get the full precision count
+
+    // Divide Uc by 26 (26MHz converted to 1MHz) todo scale for other clock freqs
+
+    Uc *= 1290555u;                  // Divide total(1/26) << 25
+    Uc >>= 25;						 // shift back.  Fixed point avoid use of floating point divide.
+    // Compiler does this inline using shifts and adds.
+
+    return Uc;
+}
+
+
+static void calc_event_counts(uint32_t timestamp)
+{
+    uint32_t calc_time, blocks, offset;
+    uint64_t aa;
+
+    calc_time = get_current_time();
+    offset = timestamp - calc_time;             // offset in useconds
+
+    if (offset > 0xf0000000u)					// if offset is a really big number, assume that timer has already expired (i.e. negative)
```

Comment:
are tabs used, the comments seem to be aligned randomly

Reply:
Probably an issue with the editor used. Will check and reformat.